services/slam/builtin/builtin.go

```diff
@@ -479,19 +479,19 @@ func NewBuiltIn(ctx context.Context, r robot.Robot, config config.Service, logge
 	var dataRate int
 	if svcConfig.DataRateMs == 0 {
 		dataRate = defaultDataRateMs
+		logger.Debugf("invalid or empty data_rate_secs given, setting to default of %d", defaultDataRateMs)
 	} else {
 		dataRate = svcConfig.DataRateMs
 	}
 
-	var mapRate int
-	if svcConfig.MapRateSec <= 0 {
-		if svcConfig.MapRateSec == -1 {
-			mapRate = 0
-		} else {
-			mapRate = defaultMapRateSec
+	mapRate := defaultMapRateSec
+	if svcConfig.MapRateSec != nil && *svcConfig.MapRateSec >= 0 {
+		if *svcConfig.MapRateSec >= 0 {
+			logger.Info("setting slam system to localization mode")
 		}
```

Comment:
The nested if statements where you check twice for the same condition are confusing. I would just go about it sequentially:
1. If the value is nil, set to the default & log it
2. Else If the value is >= 0 set to that value 
    a. If value is == 0 then log that it's in localization only mode 
3. else if the value is < 0, complain that it's an invalid number and exit (or set to default? What did we define in the tech spec?)

Reply:
moving check of map_rate being less than 0 (invalid) to configuration check similar to data_rate_msec. Nested if statement also removed